packages/dds/tree/src/test/utils.ts

```diff
@@ -61,6 +63,7 @@ export class TestTreeProvider {
     private readonly provider: ITestObjectProvider;
     private readonly _trees: ISharedTree[] = [];
     private readonly _containers: IContainer[] = [];
+    public static summarizer: Summarizer;
```

Comment:
`create()` is `static`, but this field can be non-static. Statics can be dangerous (e.g. what happens if we run tests in parallel and they share the same summarizer across tests?). 

To do this, you'll need to write a constructor for this class that takes a summarizer. I don't think constructors can be async, so it will also need to take the first tree. So you'll want two overloads, something like:

```
constructor(tree?: ISharedTree);
constructor(tree: ISharedTree, summarizer: Summarizer);
constructor(tree?: ISharedTree, summarizer?: Summarizer) {
    ...
}

Unfortunately, this also means a bunch of the code that is currently inside of the constructor will need to be lifted up into the `create()` method, since we need to be able to create the first tree before constructing, and the constructor will in fact require another parameter, namely the `TestObjectProvider` (the `provider field`).

I'm sorry this is getting more complicated than I thought it would. I think it's still very doable, but if you would like some help just let me know and we can pair program a bit.

Reply:
Updated `constructor` and `create` methods to create the first tree within `create`, and summarizer field initialized in the `constructor`